pytest_mergify/quarantine.py

```diff
@@ -0,0 +1,79 @@
+import dataclasses
+import pytest
+import _pytest.nodes
+import os
+import requests
+import typing
+
+
+def get_branch_name_for_quarantine() -> typing.Optional[str]:
+    for env_var in ("MERGIFY_TESTS_TARGET_BRANCH", "GITHUB_BASE_REF", "GITHUB_REF"):
+        env_val = os.getenv(env_var)
+        if env_val is not None:
+            return env_val
+
+    return None
+
+
+def mark_test_as_quarantined_if_needed(item: _pytest.nodes.Item) -> None:
+    if item.get_closest_marker("xfail") is None:
+        mark_test_as_quarantined(item)
+
+
+def mark_test_as_quarantined(test_item: _pytest.nodes.Item) -> None:
+    test_item.add_marker(
+        pytest.mark.xfail(
+            reason="Test is quarantined from Mergify CI Insights",
+            raises=None,
+            run=True,
+            strict=False,
+        ),
+        append=True,
+    )
```

Comment:
nit: I'd set that as staticmethod of the class to make the whole quarantine object usage transparent, but not a big deal.